### PR TITLE
Remove not needed mapper interface

### PR DESCRIPTION
**Pull request by rkondratowicz**
Interface was never really used.
Event publishers (`CreateExceptionRecord` and `AttachDocsToSupplementaryEvidence`) take concrete implementations.

More importantly, this will allow customisation of `SupplementaryEvidenceMapper` which will soon take existing documents as parameter, not just envelope from the queue.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/SupplementaryEvidenceMapper.java [hunk 1]
```diff
@@ -4,14 +4,15 @@
...
-public class SupplementaryEvidenceMapper implements ModelMapper<SupplementaryEvidence> {
+public class SupplementaryEvidenceMapper {
 
     public SupplementaryEvidenceMapper() {
         // empty mapper construct
     }
 
-    @Override
     public SupplementaryEvidence mapEnvelope(Envelope envelope) {
         return new SupplementaryEvidence(mapDocuments(envelope.documents));
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/SupplementaryEvidenceMapper.java]: As you can it only needs the documents.

After this PR is merge and interface is gone, I'll change this method to something like
```
SupplementaryEvidence map(List<Document> existingDocument, List<Document> envelopeDocuments)
```

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/DocumentMapper.java [hunk 2]
```diff
@@ -14,19 +12,21 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public interface ModelMapper<T extends CaseData> {
+public class DocumentMapper {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/DocumentMapper.java]: utility classes are final classes

**Comment by lgonczar** [comment]
I'd prefer this to be an injected dependency with instance methods, but this can be done in the future.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/DocumentMapper.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/DocumentMapper.java]: Just did a quick search on `// util class` and existing helper classes in this repo are not final.
As pmd and/or sonar don't complain, I'll leave it as is and spare myself 20min ci build waiting time ;)

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/DocumentMapper.java]: ᕕ( ᐛ )ᕗ

**Pull request merged**